swift-sdk/Internal/IterableAPIInternal.swift

```diff
@@ -326,10 +326,7 @@ final class IterableAPIInternal: NSObject, PushTrackerProtocol, AuthProvider {
     func track(inAppDelivery message: IterableInAppMessage) {
         IterableAPIInternal.call(successHandler: IterableAPIInternal.defaultOnSucess(identifier: "trackInAppDelivery"),
                                  andFailureHandler: IterableAPIInternal.defaultOnFailure(identifier: "trackInAppDelivery"),
-                                 forResult: apiClient.track(inAppDelivery: message.messageId,
-                                                            saveToInbox: message.saveToInbox,
-                                                            silentInbox: message.saveToInbox && message.trigger.type == .never,
-                                                            deviceMetadata: deviceMetadata))
+                                 forResult: apiClient.track(inAppDelivery: InAppMessageContext.from(message: message, location: .other, deviceMetadata: deviceMetadata)))
```

Comment:
Do we even need `location` for `inAppDelivery`? It will always be set to `other`, which sounds confusing since it is not a third option other than in-app or inbox, but rather that we don't know, so it should be empty.

Reply:
Actually the backend doesn't even use this parameter, `InAppDelivery` doesn't have `location` in its context.